test/functional/readpreference.test.js

```diff
@@ -655,3 +656,77 @@ describe('ReadPreference', function() {
     });
   });
 });
+
+describe.only('ReadPreference - operations', () => {
+  const COLLECTION = 'ddl_collection';
+  beforeEach(function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient(configuration.writeConcernMax(), {
+      useUnifiedTopology: true,
+      readPreference: 'primaryPreferred'
+    });
+    client.connect((err, client) => {
+      expect(err).to.not.exist;
+      const db = client.db(configuration.db);
+      db.addUser('default', 'pass', { roles: 'readWrite' }, () => {
+        db.createCollection('before_collection', () => {
+          db.createIndex(COLLECTION, { aloha: 1 }, () => {
+            client.close(done);
+          });
+        });
+      });
+    });
+  });
+
+  context('DDL methods with serverSelection readPreference primary', () => {
+    const methods = {
+      'Collection#createIndex': [{ quote: 'text' }],
+      'Db#createIndex': [COLLECTION, { quote: 'text' }],
+      'Db#addUser': ['thomas', 'pass', { roles: 'readWrite' }],
+      'Db#removeUser': ['default'],
+      'Db#createCollection': ['created_collection'],
+      'Db#dropCollection': ['before_collection'],
+      'Collection#dropIndex': ['aloha_1'],
+      'Collection#rename': ['new_name'],
+      'Db#dropDatabase': []
+    };
+    Object.keys(methods).forEach(operation => {
+      it(`should run ${operation} with serverSelection readPreference primary`, {
```

Comment:
For readability of future maintainers, I think saying something more like `should enforce fixed primary read preference`. That might even be the title of the `context`, ad the `it` should just note the operation name?